packages/jaeger-ui/src/model/ddg/DdgEVManager.test.js

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as testResources from './sample-paths.test.resources';
+import transformDdgData from './transformDdgData';
+import { changeVisibility, createVisibilityKey } from './visibility-key';
+
+import DdgEVManager from './DdgEVManager';
+
+describe('DdgEVManager', () => {
+  /**
+   * This function takes in a DdgEVManager and validates the structure based on the expected vertices.
+   *
+   * @param {DdgEVManager} ddgEVManager - The DdgEVManager to validate.
+   * @param {Object[]} expectedVertices - The vertices that the DdgEvManager should have.
+   * @param {number[]} expectedVertices[].visibilityIndices - The visibility indices that should all share one
+   *     DdgVertex.
+   * @param {number[]} expectedVertices[].egressNeighborVisibilityIndices - A single visibilityIdx is
+   *     sufficient to define a neighboring vertex. For each egressNeighborVisibilityIdx, the expectedVertex
+   *     should have an egressEdge from the expectedVertex to the vertex represented by the
+   *     egressNeighborVisibilityIdx.
+   * @param {number[]} expectedVertices[].ingressNeighborVisibilityIndices - A single visibilityIdx is
+   *     sufficient to define a neighboring vertex. For each ingressNeighborVisibilityIdx, the expectedVertex
+   *     should have an ingressEdge from the expectedVertex to the vertex represented by the
```

Comment:
`egressEdge`s are `from the vertex represented by the ingressNeighborVisibilityIdx to the expectedVertex`, this was copied and not flipped...